SDK v2 migration guide.md

```diff
@@ -56,13 +56,13 @@ to migrate to v2 when they have the chance. For more details on LTS releases, se
 | DeviceClient#registerConnectionStatusChangeCallback(IotHubConnectionStatusChangeCallback, Object); | yes | DeviceClient#setConnectionStatusChangeCallback(IotHubConnectionStatusChangeCallback, Object);  |
 | DeviceClient#setRetryPolicy(RetryPolicy); | no | DeviceClient#setRetryPolicy(RetryPolicy);  |
 | DeviceClient#getProductInfo(); | yes | DeviceClient#getProductInfo();  |
-| DeviceClient#subscribeToDeviceMethod(DeviceMethodCallback , Object, IotHubEventCallback, Object); | yes | DeviceClient#subscribeToMethodsAsync(MethodCallback, Object, IotHubEventCallback, Object);  |
```

Comment:
We had used IotHubEventCallback as the callback to notify users when most any action had completed, but it could be simplified a bit for cases like when a subscription has been acknowledged